_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,428 @@
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here:
+
+- You can reassign partitions between the brokers. For example, it can be used when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+- You can increase the number of partitions / replicas which can help with increasing the throughput of topics.
+
+## Partition Reassignment Throttle
+Reassigning partitions between brokers often leads to additional interbroker network traffic, in addition to the normal traffic required for replication.To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment. 
+This can be done using the `--throttle` flag which sets the maximum allowed bandwidth in bytes, for example `--throttle 5000000` sets the limit to 5 MB/s.
+
+Throttling might cause the reassignment to take longer to complete.
+
+1. If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+2. If the throttle is too high, the overall health of the cluster may be impacted.
+
+The best way to set the value for throttle is to start with a safe value, and if it's too low, then run the command again to update the throttle till it's good.
+
+## Actions that can be executed while using the tool
+
+It has three different actions:
+
+| Action | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file. This actions is optional if we already have `reassignment.json` file with us. You can use this actions using `--generate`|
+| `execute` | Takes a reassignment JSON file and applies it to the partitions and brokers in the cluster. The `reassignment.json` file can be either the one proposed by the `generate` action or written by the user itself. `--execute`  is used to carry out this action |
+| `verify`  | Using the same reassignment JSON file as the `--execute` step, `--verify` checks whether all the partitions in the file have been moved to their intended brokers. If the reassignment is complete, `--verify` also removes any replication quotas (`--throttle`) that are in effect. Unless removed, throttles will continue to affect the cluster even after the reassignment has finished. This action can be executed using `--verify` |
+
+## Example time
+
+Let us understand the working of this tool with a concrete example.
+Suppose we have 5 Kafka Brokers and after looking at the partition details we realize that the brokers are not too busy, so we can scale them down to 3.
```

Comment:
I know it's late but I wonder if a scaling up scenario would be more relevant

Reply:
Scale-up is something what you can also do witht he Cruise Control integration. That is not available for scale-down where manual reassignment is the only option. So the idea was to show where it is useful and where it doesn't necessarily compete with Cruise Control.

Reply:
I see, that makes sense.

Reply:
In strimzi main CC can now do scale down as well so you could discuss that as an alternative in an upcoming Strimzi release.